Experimental/CameraComponent/Viewport.cs

```diff
@@ -0,0 +1,322 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Networking.Sockets;
+using Windows.Storage;
+using Windows.UI.Composition;
+using Windows.UI.Composition.Scenes;
+using Windows.UI.Xaml;
+
+namespace CameraComponent
+{
+    public enum Stretch { Fill, FixX, FixY, Uniform, UniformToFill };
+
+    public sealed class Viewport
+    {
+        private SceneVisual sceneVisual;
+        private Vector2 size;
+        private Vector3 offset;
+        private Stretch stretch;
+        private Camera camera;
+
+        private CompositionPropertySet propertySet;
+
+        public Viewport(SceneVisual visual)
+        {
+            sceneVisual = visual;
+            var windowSize = Window.Current.Bounds;
+            size = new Vector2((float)windowSize.Width, (float)windowSize.Height);
+            //offset = 0.5f * new Vector3(size, 0.0f);
+            offset = Vector3.Zero;
+
+            stretch = Stretch.Uniform;
+        }
+
+        public SceneVisual Visual
+        {
+            get => sceneVisual;
+            set
+            {
+                sceneVisual = value;
+                CreateCameraMatrix();
+            }
+        }
+        public Vector2 Size
+        {
+            get
+            {
+                if (camera.UseAnimations)

```

Comment:
TODO: a discussion about now that expression animations are working, whether we should remove the non-expression code paths completely.